### PR TITLE
chore(frontend): Upgrade to React Query v5

### PR DESCRIPTION
**Pull request by lhvy**

### CODE REVIEW

File: frontend/src/pages/CourseSelector/CourseSelector.tsx [hunk 1]
```diff
@@ -70,7 +74,7 @@ const CourseSelector = () => {
     };
     resizerDiv?.addEventListener('mousedown', startResize);
     return () => resizerDiv?.removeEventListener('mousedown', startResize);
-  }, []);
+  }, [showedNotif, coursesQuery.isSuccess, coursesQuery.data]);
```
Comment (ollibowers) [inline frontend/src/pages/CourseSelector/CourseSelector.tsx]: could we put the notification in its own useEffect? keeping them compartmentalized when they are doing different things (and have different dependencies) is good

File: frontend/src/pages/TermPlanner/DraggableCourse/DraggableCourse.tsx
Comment (ollibowers) [inline frontend/src/pages/TermPlanner/DraggableCourse/DraggableCourse.tsx:1]: hmmmm do we want these diffs?

**Review by ollibowers — CHANGES_REQUESTED** [review]

File: frontend/src/pages/CourseSelector/CourseSelector.tsx (hunk 1 above)
Comment (lhvy) [reply frontend/src/pages/CourseSelector/CourseSelector.tsx]: Fixed

File: frontend/src/pages/TermPlanner/DraggableCourse/DraggableCourse.tsx
Comment (lhvy) [reply frontend/src/pages/TermPlanner/DraggableCourse/DraggableCourse.tsx:1]: Fixed

**Review by ollibowers — APPROVED** [review]
looks good

**Pull request merged**